rabbitmq_binding: Fix using empty routing key

sn4kebite opened the pull request:
If the routing key is an empty string we need to use the special props
value ~ in the URL. Otherwise the last part of the URL will be empty,
which will instead be equivalent to the "list all bindings" API call
which always returns 200 OK (as long as the exchange and queue exists).

##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
rabbitmq_binding

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes -->
```paste below
ansible 2.8.0.dev0 (devel 03f71e778b) last updated 2018/11/13 13:28:04 (GMT +200)
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/jon/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/jon/ansible-git/lib/ansible
  executable location = /usr/bin/ansible
  python version = 3.7.1 (default, Oct 22 2018, 10:41:28) [GCC 8.2.1 20180831]
```

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

Example playbook:
```yaml
- hosts: localhost
  connection: local
  tasks:
    - name: rabbitmq exchange
      rabbitmq_exchange:
        name: test
        type: direct
    - name: rabbitmq queue
      rabbitmq_queue:
        name: test
    - name: rabbitmq bindings
      rabbitmq_binding:
          source: test
          destination: test
          routing_key: ''
          type: queue
```

Before fix (no changes are detected even when the binding is missing):
```paste below
PLAY [localhost] **************************************************************************************************************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] ********************************************************************************************************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [rabbitmq exchange] ******************************************************************************************************************************************************************************************************************************************************************************************************
changed: [localhost]

TASK [rabbitmq queue] *********************************************************************************************************************************************************************************************************************************************************************************************************
changed: [localhost]

TASK [rabbitmq bindings] ******************************************************************************************************************************************************************************************************************************************************************************************************
ok: [localhost]

PLAY RECAP ********************************************************************************************************************************************************************************************************************************************************************************************************************
localhost                  : ok=4    changed=2    unreachable=0    failed=0   
```

After fix:
```paste below
 PLAY [localhost] **************************************************************************************************************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] ********************************************************************************************************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [rabbitmq exchange] ******************************************************************************************************************************************************************************************************************************************************************************************************
changed: [localhost]

TASK [rabbitmq queue] *********************************************************************************************************************************************************************************************************************************************************************************************************
changed: [localhost]

TASK [rabbitmq bindings] ******************************************************************************************************************************************************************************************************************************************************************************************************
changed: [localhost]

PLAY RECAP ********************************************************************************************************************************************************************************************************************************************************************************************************************
localhost                  : ok=4    changed=3    unreachable=0    failed=0    skipped=0   
```

sn4kebite committed:
rabbitmq_binding: Fix using empty routing key

If the routing key is an empty string we need to use the special props
value ~ in the URL. Otherwise the last part of the URL will be empty,
which will instead be equivalent to the "list all bindings" API call
which always returns 200 OK (as long as the exchange and queue exists).

Shaps reviewed (changes requested):
The `routing_key` parameter defaults to `#` [(L279)](https://github.com/ansible/ansible/blob/a02558c8ff8a9eca65a25485f5b3137c34ab7690/lib/ansible/modules/messaging/rabbitmq/rabbitmq_binding.py#L279) so I don't see how it could be an empty string. In  any case I would rather change the `default` then have an `if ... else ...`

sn4kebite commented:
Just changing the default won't work. If you explicitly specify an empty string (like in the provided example) as the routing key it will still break.

Shaps commented:
This is a fair point, I'd still prefer to have the `if/else` outside the `format` setting the `self.routing_key` var

sn4kebite committed:
rabbitmq_binding: Move routing_key test outside format

Test the routing_key value and set the props value outside the format
call. We leave the original routing_key untouched since its original
value is later needed in create().

Shaps commented:
shipit

resmo reviewed (approved):
LGTM
